pkg/server/sessions/manager.go

```diff
@@ -0,0 +1,244 @@
+/*
+Copyright 2021 CodeNotary, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sessions
+
+import (
+	"context"
+	"github.com/codenotary/immudb/pkg/auth"
+	"github.com/codenotary/immudb/pkg/errors"
+	"github.com/codenotary/immudb/pkg/logger"
+	"github.com/codenotary/immudb/pkg/server/sessions/internal/transactions"
+	"github.com/rs/xid"
+	"os"
+	"sync"
+	"time"
+)
+
+var ErrGuardAlreadyRunning = errors.New("session guard already launched")
+var ErrGuardNotRunning = errors.New("session guard not running")
+
+var guard *manager
+
+type manager struct {
+	Running    bool
+	sessionMux sync.Mutex
+	guardMux   sync.Mutex
+	sessions   map[string]*Session
+	ticker     *time.Ticker
+	done       chan bool
+	logger     logger.Logger
+	options    *Options
+}
+
+type Manager interface {
+	NewSession(user *auth.User, databaseID int64) string
+	SessionPresent(sessionID string) bool
+	DeleteSession(sessionID string) error
+	UpdateSessionActivityTime(sessionID string)
+	UpdateHeartBeatTime(sessionID string)
+	StartSessionsGuard() error
+	StopSessionsGuard() error
+	GetSession(sessionID string) (*Session, error)
+	CountSession() int
+	GetTransactionFromContext(ctx context.Context) (transactions.Transaction, error)
+	DeleteTransaction(transactions.Transaction) error
+}
+
+func NewManager(options *Options) *manager {
```

Comment:
just to be consistent with other parts of the source code where it's options is expected not to be nil and and error is returned in such case or even when provided options are not valid

Reply:
it's something we can improve... probably not using a pointer but the plain options struct ...